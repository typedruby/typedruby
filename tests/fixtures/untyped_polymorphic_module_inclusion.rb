# intentionally no typedruby annotation here

class Foo
  include Enumerable
end

def main => nil
  Foo.new.map { |x| x }
end
