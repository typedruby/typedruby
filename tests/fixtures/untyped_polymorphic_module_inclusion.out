error: Could not match types:

        @ __ROOT__/tests/fixtures/untyped_polymorphic_module_inclusion.rb:7
      7 |  def main => nil
                       ^^^ NilClass, with:
        @ (builtin stdlib):__LINE__
    107 |    def map[ProjectedType]({ |EnumType x| => ProjectedType } &) => [ProjectedType]; end
                                                                            ^^^^^^^^^^^^^^^ Array::[:any]

        - arising from an attempt to match:

        @ __ROOT__/tests/fixtures/untyped_polymorphic_module_inclusion.rb:7
      7 |  def main => nil
                       ^^^ NilClass, with:
        @ __ROOT__/tests/fixtures/untyped_polymorphic_module_inclusion.rb:8
      8 |    Foo.new.map { |x| x }
             ^^^^^^^^^^^^^^^^^^^^^ Array::[:any]
